src/matrix_cookbook/mat_eq27_lib.lean

```diff
@@ -0,0 +1,206 @@
+import linear_algebra.matrix.trace
+import linear_algebra.matrix.determinant
+import data.matrix.notation
+import data.fintype.big_operators
+import tactic.ring
+import tactic.norm_fin
+import data.is_R_or_C.basic
+import algebra.char_p.basic
+
+open_locale matrix big_operators
+open matrix
+
+variables {R : Type*}[field R][char_zero R]
+
+lemma trace_a_fin4 {A : matrix (fin 4) (fin 4) R} :
+  A.trace = A 0 0 + A 1 1 + A 2 2 + A 3 3 := 
+begin
+  rw matrix.trace,  rw fin.sum_univ_four, repeat {rw diag,},
+end
+
+lemma trace_a_squared {A : matrix (fin 4) (fin 4) R}:
+  trace (A^2) = 
+    A 0 0^2 + A 1 1^2 + A 2 2^2 + A 3 3^2 + 
+    2*A 0 1*A 1 0 + 2*A 0 2*A 2 0 + 2*A 0 3*A 3 0 + 
+    2*A 1 2*A 2 1 + 2*A 1 3*A 3 1 + 2*A 2 3*A 3 2 := 
+begin
+  rw matrix.trace,
+  simp only [diag],
+  rw pow_two, rw mul_eq_mul, rw fin.sum_univ_four,
+  repeat{rw mul_apply, rw fin.sum_univ_four},
+  ring,
+end
+
+lemma trace_a_cubed {A : matrix (fin 4) (fin 4) R}:
```

Comment:
Generally lemma names shouldn't refer to the names of the variables they use; so this one would be

```suggestion
lemma trace_pow_three_fin_four {A : matrix (fin 4) (fin 4) R}:
```